src/Xamarin.Android.Build.Tasks/Tasks/GenerateTypeMappings.cs

```diff
@@ -75,9 +78,16 @@ void WriteTypeMappings (NativeCodeGenState state)
 	{
 		if (androidRuntime == Xamarin.Android.Tasks.AndroidRuntime.NativeAOT) {
 			// NativeAOT typemaps are generated in `Microsoft.Android.Sdk.ILLink.TypeMappingStep`
+			Log.LogDebugMessage ("Skipping type maps for NativeAOT.");
 			return;
 		}
 		Log.LogDebugMessage ($"Generating type maps for architecture '{state.TargetArch}'");
+
+		if (TypemapImplementation != "llvm-ir") {
+			Log.LogDebugMessage ($"TypemapImplementation='{TypemapImplementation}' will write an empty native typemap.");
+			state = new NativeCodeGenState (state.TargetArch, new TypeDefinitionCache (), state.Resolver, [], [], state.Classifier);
+		}
```

Comment:
Instead of returning early, I made this write the native assembly with a typemap of 0 entries, this prevents the crash at runtime:
```
03-12 12:36:53.682 29155 29155 E AndroidRuntime: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "java_type_count" referenced by "/data/app/~~sPVK03Ot__92Sd7QEgoNXw==/com.companyname.helloandroid-EN9cYvEYQtDuCTIXF_jW0Q==/split_config.arm64_v8a.apk!/lib/arm64-v8a/libmonodroid.so"...
```